src/bluesearch/database/article.py

```diff
@@ -73,6 +76,121 @@ def paragraphs(self) -> Iterable[Tuple[str, str]]:
         """
 
 
+class PubmedXMLParser(ArticleParser):
+    """Parser for PubMed XML files using the JATS Journal Publishing DTD.
+
+    Parameters
+    ----------
+    path
+        The path to the XML file from PubMed.
+    """
+
+    def __init__(self, path: Union[str, Path]) -> None:
+        self.content = etree.parse(str(path))  # nosec
```

Comment:
To actually use inheritance, `super().__init__(...)` should be called.

This is also a best practice to avoid the current case. Indeed, the child classes have a different API. This is a contradiction with the goal of using inheritance.

Reply:
Done in 098c744

Reply:
The complete implementation will be part of https://github.com/BlueBrain/Search/issues/419.

Indeed:
- the API of parser classes will be harmonized,
- there will be a constructor in the parent class,
- it will be called through `super()`.